velox/exec/fuzzer/WindowFuzzer.cpp

```diff
@@ -53,8 +54,56 @@ void WindowFuzzer::addWindowFunctionSignatures(
 }
 
 const std::string WindowFuzzer::generateFrameClause() {
-  // TODO: allow randomly generating the frame clauses.
-  return "RANGE BETWEEN UNBOUNDED PRECEDING AND CURRENT ROW";
+  auto frameType = [](int value) -> const std::string {
+    switch (value) {
+      case 0:
+        return "ROWS";
+      case 1:
+        return "RANGE";
+      default:
+        VELOX_UNREACHABLE("Unknown value for frame bound generation");
+    }
+  };
+  auto frameTypeValue =
+      boost::random::uniform_int_distribution<uint32_t>(0, 1)(rng_);
+  auto frameTypeString = frameType(frameTypeValue);
+
+  auto frameBound = [&](int value, bool start) -> const std::string {
+    // Generating only constant bounded k PRECEDING/FOLLOWING frames for now.
+    auto kValue =
+        boost::random::uniform_int_distribution<uint32_t>(1, 10)(rng_);
+    switch (value) {
+      case 0:
+        return "CURRENT ROW";
+      case 1:
+        return start ? "UNBOUNDED PRECEDING" : "UNBOUNDED FOLLOWING";
+      case 2:
+        return fmt::format("{} FOLLOWING", kValue);
+      case 3:
+        return fmt::format("{} PRECEDING", kValue);
+      default:
+        VELOX_UNREACHABLE("Unknown option for frame clause generation");
+    }
+  };
+
+  // Generating k PRECEDING and k FOLLOWING frames only for ROWS type.
+  // k RANGE frames require more work as we have to generate columns with the
+  // frame bound values.
+  auto startValue = boost::random::uniform_int_distribution<uint32_t>(
+      0, frameTypeValue == 0 ? 3 : 1)(rng_);
```

Comment:
Add a lambda auto isRowsFrame = [](uint32_t frameTypeValue) { return frameTypeValue == 0; } and use that in the logic to make it more readable.

Reply:
Removed this check and start/end bounds are chosen by picking a random index from start and end bound options.